SD_Assignment1/src/main/java/ro/utcn/aron/memory/InMemoryQuestionRepository.java

```diff
@@ -0,0 +1,76 @@
+package ro.utcn.aron.memory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+import ro.utcn.aron.api.QuestionRepository;
+import ro.utcn.aron.model.Question;
+
+@Component
+@ConditionalOnProperty(name = "aron.repository-type", havingValue = "MEMORY")
+public class InMemoryQuestionRepository implements QuestionRepository {
+
+	private AtomicInteger currentId = new AtomicInteger(0);
+	private Map<Integer, Question> data = new ConcurrentHashMap<>();
+	
+	@Override
+	public synchronized Question save(Question question) {
+		if(question.getId() != 0) {
+			data.put(question.getId(), question);
+		} else {
+			question.setId(currentId.get());
+			currentId.set(currentId.get()+1);
+			data.put(question.getId(), question);
+		}
+		return question;
+	}
+
+	@Override
+	public Optional<Question> findByID(int id) {
+		return Optional.ofNullable(data.get(id));
+	}
+
+	@Override
+	public synchronized void remove(Question question) {
+		data.remove(question.getId());
+		
+	}
+
+	@Override
+	public List<Question> findAll() {
+		return new ArrayList<>(data.values());
+	}
+
+	@Override
+	public Optional<List<Question>> filterByTitle(String title) {
+		List<Question> result = new ArrayList<>();
+		data.forEach((K, V)->{
+			if(title.contains(V.getTags())) {

```

Comment:
This doesn't actually filter by title, but it checks if the given title contains the tags...